site/gatsby-site/gatsby-node.js

```diff
@@ -50,6 +51,77 @@ exports.createPages = ({ graphql, actions }) => {
     );
   });
 
+  const wordCounts = new Promise((resolve, reject) => {
```

Comment:
Can we put this inside the wordcount component, or some other codepoint close to it and then require the functionality into here? I don't want to bloat gatsby-node or make it more difficult to maintain the component. I also want to establish a design pattern where we can do this design pattern many times.

Reply:
I agree to not make the gatsby-node giant, let me make few functions to do this outside. Give me 30 mins, I'll update the PR.

Reply:
Sounds good, thanks!

Reply:
I'm going to create a `page-creators` directory at the top level, and will put `mdx.js`, `citiation.js`, `wordCounts.js` which returns a function accept `createPage` and `graphql` as props and return a promise. 

so createPages will look like

```
const createMdxPages = require('./page-creators/createMdxPages');
const createCitiationPages = require('./page-creators/createCitiationPages');
const createWordCountsPages = require('./page-creators/createWordCountsPage');

exports.createPages = ({ graphql, actions }) => {

  const { createPage } = actions;

  return Promise.all([
    createMdxPages(graphql, createPage),
    createCitiationPages(graphql, createPage),
    createWordCountsPages(graphql, createPage),
  ])
};
```